compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1010,13 +1010,25 @@ object Parsers {
       skipParams()
       lookahead.isColon
       && {
+        lookahead.nextToken()
```

Comment:
I would move the `lookahead.nextToken()` to line 1018, just before
`!lookahead.isAfterLineEnd`. That saves us a useless `nextToken` when source is 3.6.